R/data-set.R

```diff
@@ -51,76 +60,166 @@ DataSet <- R6::R6Class(
       }
       private$.yValues$displayUnit <- value
     },
-    #' @field yValues Values stored in the yUnit dimension (not necessarily in the base unit of the dimension)
+    #' @field yValues Values stored in the yDimension dimension (not necessarily in the base unit)
     yValues = function(values) {
       if (missing(values)) {
-        return(private$getColumnValues(private$.yValues))
+        return(private$.getColumnValues(private$.yValues))
+      }
+      validateIsSameLength(self$xValues, values)

```

Comment:
We have to find a proper way to synchronize the lengths of base grid and data columns, it's pure pain!

Reply:
Make sure that x and y values always have the same length. Will be covered by setting xValues yValues yErrorValues to readonly and implementing a separate method to set values.